WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -2658,6 +2717,46 @@ function WeakAuras.EnsureOptions(id)
   end
 end
 
+local scheduleRefillFrame = CreateFrame("FRAME", "", UIParent)
+scheduleRefillFrame.OnUpdate = function()
+  for id, data in pairs(scheduleRefillFrame.ids) do
+    if not(displayOptions[id]) then
+      WeakAuras.AddOption(id, data);
+    end
+  end
+  wipe(scheduleRefillFrame.ids)
+
+  frame:RefillOptions()
+  frame:SetScript("OnUpdate", nil)
+end
+
+scheduleRefillFrame.ScheduleRefillOnly = function()
+  frame:SetScript("OnUpdate", scheduleRefillFrame.OnUpdate)
+end
+scheduleRefillFrame.ScheduleReload = function(self, id, data)
+  scheduleRefillFrame.ids[id] = data
+  scheduleRefillFrame:ScheduleRefillOnly()
+end
+
+scheduleRefillFrame.ids = {}
+
+
+-- TODO replace older ReloadOptions, SchduleReloadOptions, ReloadTriggerOptions, ReloadGroupRegionOptions
+-- automatically clear parent/tempGroup ?
```

Comment:
This is something that I want to tackle soon, and will rework how we treat the options.

Basically, we sometimes need to regenerate the options table, and sometimes just tell AceGui that our options changed even without the table being changed.

Currently we do that by immediately recreating a new options table. What I want to try out, is to split that up into two steps:

a) Just mark which options tables need to be recreated / mark that options need to be refilled
b) In a OnUpdate handler recreate the options table.

This should ensure that each change only recreates each table once at most. I have to test and remove lots of old code, but that'S the basic idea. I think that should make editing big groups much faster.

Reply:
Can you speak about that in #821?